include/dca/phys/dca_step/cluster_solver/ctint/structs/solver_configuration.hpp

```diff
@@ -0,0 +1,391 @@
+// Copyright (C) 2018 ETH Zurich
+// Copyright (C) 2018 UT-Battelle, LLC
+// All rights reserved.
+//
+// See LICENSE.txt for terms of usage.
+// See CITATION.md for citation guidelines, if DCA++ is used for scientific publications.
+//
+// Authors: Giovanni Balduzzi (gbalduzz@phys.ethz.ch)
+//
+// This class organizes the vertex configuration for ct-int.
+
+#ifndef DCA_PHYS_DCA_STEP_CLUSTER_SOLVER_CTINT_STRUCTS_SOLVER_CONFIGURATION_HPP
+#define DCA_PHYS_DCA_STEP_CLUSTER_SOLVER_CTINT_STRUCTS_SOLVER_CONFIGURATION_HPP
+
+#include <array>
+#include <numeric>
+#include <vector>
+
+#include "dca/io/buffer.hpp"
+#include "dca/phys/dca_step/cluster_solver/ctint/structs/ct_int_matrix_configuration.hpp"
+#include "dca/phys/dca_step/cluster_solver/ctint/structs/interaction_vertices.hpp"
+#include "dca/phys/dca_step/cluster_solver/ctint/structs/utils.hpp"
+#include "dca/linalg/device_type.hpp"
+
+namespace dca {
+namespace phys {
+namespace solver {
+namespace ctint {
+// dca::phys::solver::ctint::
+
+class SolverConfiguration : public MatrixConfiguration {
+public:
+  using BaseClass = MatrixConfiguration;
+
+  SolverConfiguration() {}
+  SolverConfiguration(const SolverConfiguration& other) = default;
+  SolverConfiguration(SolverConfiguration&& other) = default;
+  SolverConfiguration(double beta, int n_bands, const InteractionVertices& H_int,
+                      double double_move_prob = 0);
+
+  SolverConfiguration& operator=(const SolverConfiguration& other) = default;
+  SolverConfiguration& operator=(SolverConfiguration&& other) = default;
+
+  bool operator==(const SolverConfiguration& rhs) const;
+
+  template <class RngType>
+  void insertRandom(RngType& rng);
+
+  // TODO: pass output as argument.
+  template <class RngType>
+  std::vector<int> randomRemovalCandidate(RngType& rng, double removal_rand);
+  template <class RngType>
+  std::vector<int> randomRemovalCandidate(RngType& rng);
+
+  // Remove elements with id remove by copying elements from the end.
+  // Postcondition: from and sector_from contains the indices that where moved into the place
+  //                of the old elements.
+  // Postcondition: all vectors are ordered, resized, and ready to be used for moving the rows of
+  //                the M matrix
+  // In/Out: sector_remove, remove.
```

Comment:
These are probably better variable names. But change them here or in the declaration they need to match.

Reply:
Changed name of  the arguments to match the documentation in 35b02bc5a9f92c7d3a74ad038a94b299ee0f3fe1